workers/check_for_new_commits_on_github_worker.rb

```diff
@@ -1,78 +1,28 @@
-require_relative "worker_base"
+require_relative "github_worker_base"
 require_relative "worker_scheduler"
 require_relative "../services/build_service"
-require_relative "../shared/models/provider_credential"
 require_relative "../shared/models/job_trigger"
 require_relative "../shared/logging_module"
-require_relative "../services/test_runner_service"
-require_relative "../services/code_hosting/git_hub_service"
 
 module FastlaneCI
   # Responsible for checking if there have been new commits
   # We have to poll, as there is no easy way to hear about
   # new commits from web events, as the CI system might be behind
   # firewalls
-  class CheckForNewCommitsOnGithubWorker < WorkerBase
+  class CheckForNewCommitsOnGithubWorker < GithubWorkerBase
     include FastlaneCI::Logging
 
-    attr_accessor :provider_credential
-    attr_accessor :project
-
-    attr_accessor :user_config_service
-    attr_accessor :github_service
-
-    attr_accessor :serial_task_queue
+    attr_accessor :trigger_type
     attr_accessor :current_tasks
-    attr_accessor :target_branches_set
-
-    attr_writer :git_repo
-
-    def provider_type
-      return FastlaneCI::ProviderCredential::PROVIDER_CREDENTIAL_TYPES[:github]
-    end
+    attr_accessor :scheduler
 
     def initialize(provider_credential: nil, project: nil)
-      self.provider_credential = provider_credential
-      self.github_service = FastlaneCI::GitHubService.new(provider_credential: provider_credential)
-      self.project = project
-
-      self.target_branches_set = Set.new
-      project.job_triggers.each do |trigger|
-        if trigger.type == FastlaneCI::JobTrigger::TRIGGER_TYPE[:commit]
-          self.target_branches_set.add(trigger.branch)
-        end
-      end
+      self.trigger_type = FastlaneCI::JobTrigger::TRIGGER_TYPE[:commit]
+      self.scheduler = WorkerScheduler.new(sleep_interval: 10)
 
-      project_full_name = project.repo_config.git_url
-
-      if project.repo_config.kind_of?(FastlaneCI::GitRepoConfig)
-        project_full_name = project.repo_config.full_name unless project.repo_config.full_name.nil?
-      end
-
-      self.serial_task_queue = TaskQueue::TaskQueue.new(name: "#{project_full_name}:#{provider_credential.ci_user.email}")
       self.current_tasks = []
 
-      super() # This starts the work by calling `work`
```

Comment:
will still need to call super with `super(provider_credential: provider_credential, project: project)`

Reply:
Yes. I should do this. 
I did this originally but then found out without the () super gets called with all the same args

Reply:
If you want to keep it as `super` I'd be ok with that. Originally I made this comment because it didn't look like you called `super` at all due to how GitHub had everything highlighted for me.
I usually prefer explicitly calling methods with named params, but I think the way you have it might be more idiomatic to ruby, so either way 👍

Reply:
done